CRM/Upgrade/Incremental/php/FiveThirtyFour.php

```diff
@@ -70,10 +94,56 @@ public function upgrade_5_34_alpha1(string $rev): void {
 
     $this->addTask('core-issue#365 - Add effective_end_date to civicrm_action_schedule', 'addColumn',
       'civicrm_action_schedule', 'effective_end_date', "timestamp NULL COMMENT 'Latest date to consider end events from.'");
+
+    if (extension_loaded('mcrypt') && !empty(self::findSmtpPasswords())) {
+      $this->addTask('Migrate SMTP password', 'migrateSmtpPasswords');
+    }
   }
 
   // public static function taskFoo(CRM_Queue_TaskContext $ctx, ...) {
   //   return TRUE;
   // }
 
+  /**
+   * @return array
+   *   A list of "civicrm_setting" records which have
+   *   SMTP passwords, or NULL.
+   */
+  protected static function findSmtpPasswords() {
+    $query = CRM_Utils_SQL_Select::from('civicrm_setting')
+      ->where('name = "mailing_backend"');
+
+    $matches = [];
+    foreach ($query->execute()->fetchAll() as $setting) {
+      $value = unserialize($setting['value']);
+      if (!empty($value['smtpPassword'])) {
+        $matches[] = $setting;
+      }
+    }
+
+    return $matches;
+  }
+
+  /**
+   * Find any SMTP passwords. Remove the CRM_Utils_Crypt encryption.
+   *
+   * Note: This task is only enqueued if mcrypt is active.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function migrateSmtpPasswords(CRM_Queue_TaskContext $ctx) {
+    $settings = self::findSmtpPasswords();
+    foreach ($settings as $setting) {
+      $value = unserialize($setting['value']);
+      $value['smtpPassword'] = CRM_Utils_Crypt::decrypt($value['smtpPassword']);
```

Comment:
Is there a reason why you don't want to call crypto service and encrypt using the new system here?

Reply:
For posterity, we discussed further at https://chat.civicrm.org/civicrm/pl/r8pbskapc7gybngdceqxg9sksh

Basically, we're evaluating trade-off between:

* SITE_KEY as fallback crypto key
    * __How__: Register `SITE_KEY` as low-priority crypto-key for use with `CRED`. The upgrader uses this for initial cryption of `smtpPassword`
    * __Pro__: The `smtpPassword` is promptly (re)encrypted. No interregnum.
    * __Con__: Creates more edge-cases / complexity
* Temporary decryption
    * __How__: The upgrader decrypts `smtpPassword`. After upgrade, sysadmin defines CRED_KEYS and runs `rotateKey`
    * __Pro__: Long-term, system is simpler to understand / has fewer edge-cases (for admin/dev/etc)
    * __Con__: There's an interregnum (between the upgrade and encryption) during which `smtpPassword` is stored as plaintext.